lib/src/main/java/org/team1540/rooster/drive/pipeline/SimpleJoystickInput.java

```diff
@@ -0,0 +1,132 @@
+package org.team1540.rooster.drive.pipeline;
+
+import edu.wpi.first.wpilibj.Joystick;
+import java.util.OptionalDouble;
+import org.team1540.rooster.Utilities;
+
+/**
+ * Simple tank-style input from a WPILib {@link Joystick}. The left and right joysticks are used to
+ * control the left and right sides of the robot respectively. Additionally, there is an optional
+ * "forward axis" and "back axis"; if the forward axis is bound to one trigger on an XBox controller
+ * and the back to the other, the forward trigger will cause the robot to drive straight forward and
+ * the back trigger will cause it to drive straight backwards. This {@link Input} creates a {@link
+ * TankDriveData} with only the feed-forward fields set on the left and right sides.
+ */
+public class SimpleJoystickInput implements Input<TankDriveData> {
+
+  private Joystick joystick;
+  private int leftAxis;
+  private int rightAxis;
+  private int fwdAxis;
+  private int backAxis; // or "baxis" if you don't have much time
+  private boolean invertLeft;
+  private boolean invertRight;
+
+  private double deadzone;
+
+  @Override
+  public TankDriveData get() {
+    double triggerValue;
+    if (fwdAxis != -1 && backAxis != -1) {
+      triggerValue = Utilities.processDeadzone(joystick.getRawAxis(fwdAxis), deadzone)
+          - Utilities.processDeadzone(joystick.getRawAxis(backAxis), deadzone);
+    } else {
+      triggerValue = 0;
+    }
+    double leftThrottle = Utilities.constrain(
+        Utilities.processDeadzone(Utilities.invertIf(invertLeft, joystick.getRawAxis(leftAxis)),
+            deadzone) + triggerValue,
+        1
+    );
+    double rightThrottle = Utilities.constrain(
+        Utilities.processDeadzone(Utilities.invertIf(invertRight, joystick.getRawAxis(rightAxis)),
+            deadzone) + triggerValue,
+        1
+    );
+
+    return new TankDriveData(
```

Comment:
so i am confusion

Reply:
Ewww yeah what did IntelliJ auto format do to that; I’ll clean it up

Reply:
should hopefully be a bit better